### ISSUE TITLE
[FeatureRequest] Buffer or UInt8Array for input

### ISSUE THREAD

**Issue by jedwards1211** [antelle/node-stream-zip#99]
Someone was saying in #97 that consuming a stream isn't really feasible because metadata is at the end of the file.  But it would be nice to be able to at least read a `Buffer`, because I'd like to use this to read small zip files sent via HTTP form upload without having to save them to disk.

**Comment by gowin20** [comment]
@antelle any plans to add `Buffer` support? I am creating a large zip of images using [sharp](https://sharp.pixelplumbing.com/api-output#tile) and need to access them directly without saving to disk.

**Comment by simonpai** [comment]
There is a way to work around. The idea is quite simple:

1. Use `StreamZip.setFs()` to make it use a custom `fs` internally.
2. Use `memfs` to write a `Buffer` as a file.

Here's the sample code:

```js
import StreamZip from 'node-stream-zip';
import { createFsFromVolume, Volume } from 'memfs';
import { Union } from 'unionfs';

// setup
const volume = new Volume();
const memFs = createFsFromVolume(volume).promises;
const ufs = new Union();
ufs.use(fs).use(volume);
let memFileIndex = 0;
StreamZip.setFs(ufs);

async function readZipFromBuffer(buffer, callback) {
  const file = `/file_${memFileIndex++}.zip`;
  let zip;
  try {
    await memFs.writeFile(file, buffer);
    zip = new StreamZip.async({ file });
    return await callback(zip);
  } finally {
    if (zip) {
      await zip.close();
    }
    await memFs.unlink(file);
  }
}

// usage
const data = await readZipFromBuffer(buffer, async (zip) => {
  // ... do whatever with zip
  // return ...
});
```

Some references:
* [unionfs](https://github.com/streamich/unionfs)
* [memfs](https://github.com/streamich/memfs/blob/master/docs/node/reference.md)

**Comment by jedwards1211** [comment]
That's a pretty drastic workaround...I'd prefer to be able to pass a buffer directly to `node-stream-zip`.  Clever idea though

**Comment by simonpai** [comment]
> That's a pretty drastic workaround...I'd prefer to be able to pass a buffer directly to `node-stream-zip`. Clever idea though

Yeah, I hope for the same thing but this project seems to be inactive for a few years now. Alternatively we could've forked the project and make it accept a Buffer, yet I believe an implementation with the same `memfs` approach would still be the shortest path given how much `node-stream-zip` is coupled with `fs`.
